docs/architecture.md

```diff
@@ -65,102 +70,109 @@ Design constraints are:
+
+7. SPDX-Builder writes an SBOM report in valid SPDX format.
+
+8. SPDX-Builder (optionally) uploads the SPDX file to an archive.
+
+#### Variant: Source is OSS Review Toolkit Analyzer
 
-SPDX-Builder is a command-line application that converts the output of
-the [OSS Review Toolkit (ORT)](https://github.com/oss-review-toolkit/ort)
-Analyzer tool into a SPDX bill-of-materials report in tag-value format, while (
-optionally)
-merging license information scanned from package source code by an external
-knowledge base like
-the [License Scanner service](https://github.com/philips-software/license-scanner)
-.
+The Analyzer component
+of [OSS Review Toolkit](https://github.com/oss-review-toolkit/ort)
+uses the build environment to retrieve detailed package metadata from the
+backing repository and other locations, yielding an SBOM in its proprietary
+format.
 
-### Bill-of-materials
+In this case the knowledge base consists of
+the [License Scanner Service](https://github.com/philips-softwre/license-scanner)
+(or [BOM-Base](https://github.com/philips-softwre/bom-base) as compatible
+alternative) to add licenses scanned from the source files when the source
+location is available from the metadata provided by ORT.
 
-The domain for this application consists of the classes and relations depicted
-in the figure below:
+If the declared license provided by ORT does not match the scanned license from
+the knowledge base, this licenses is "contested" to the knowledge base. If the
+knowledge base indicates the scanned license is "confirmed", SPDX-Builder uses
+it as the concluded license (instead of the defaulting to the declared license).
 
-![UML class diagram](domain.png "Domain classes modeling a bill-of-materials")
+#### Variant: Source is Black Duck
 
-A `BillOfMaterials` is a container that holds `Package`s that are connected by
-typed `Relation`s. All other information is represented in attibutes of these
-domain classes.
+The "hierarchical BOM" API of Black Duck (if enabled) exposes packages and their
+relations via an online API that is queried by SPDX-Builder using an
+authentication token.
 
-### Integration of license scan results
+As Black Duck provides an integrated SBOM management environment and knowledge
+base, the package metadata is not enhanced by an external knowledge base.
 
-If the URL to a license scanner is specified, the source code location (if
-available) of every package is submitted for scanning. The license scanner
-immediately responds with available license information, or else schedules the
-package for scanning later. This leads to two licenses per package:
+#### Variant: Source is a textual package tree
 
-- "Declared license" is the license provided in the metadata from the package
-  manager.
-- "Detected license" is the license independently scanned from the package
-  source code files.
+To reduce coupling with the build environment of projects, a textual tree of
```

Comment:
Maybe also mention two extra things:

1. Asking the Application Expert to come-up with a basic dependency tree, will increase the likelihood the SBOM is correct. An Application Expert knows the application and immediately excludes dependencies which are not part of the actual product f.e. Examples, documentation or Build-tools.
2. We think this is the preferred way

Reply:
Added note about the advantage of having a developer configure a known build tool.

(I don't think that our _preliminary_ assumption this could be the way forward should be mentioned in this tool implementation.)